File: lib/models/driver.dart
class Driver {
  int? driverId;
  String? bikeNo;
  String? phoneNo;
  String? name;
  String? password;
  String? profilePic;
  String? gender;
  String vehicleType;

  Driver({
    this.driverId,
    this.bikeNo,
    this.phoneNo,
    this.name,
    this.password,
    this.profilePic,
    this.gender,
    required this.vehicleType,
  });

  factory Driver.fromJson(Map<String, dynamic> json) {
    return Driver(
      driverId: json['driver_id'],
      bikeNo: json['bike_no'],
      phoneNo: json['phone_no'],
      name: json['name'],
      password: json['password'],
      profilePic: json['profile_pic'],
      gender: json['gender'],
      vehicleType: json['vehicle_type'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'driver_id': driverId,
      'bike_no': bikeNo,
      'phone_no': phoneNo,
      'name': name,
      'password': password,
      'profile_pic': profilePic,
      'gender': gender,
      'vehicle_type': vehicleType,
    };
  }
}

File: cd-creations-helpme-web-backend/models/driver.js
// models/driver.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Driver = sequelize.define('Driver', {
  driver_id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  bike_no: {
    type: DataTypes.STRING,
  },
  phone_no: {
    type: DataTypes.STRING,
  },
  name: {
    type: DataTypes.STRING,
  },
  password: {
    type: DataTypes.STRING,
  },
  profile_pic: {
    type: DataTypes.STRING,
  },
  gender: {
    type: DataTypes.STRING,
  },
  vehicle_type: {
    type: DataTypes.STRING(50),
    allowNull: true,
  }
}, {
  tableName: 'driver', // Specify the table name explicitly to match your existing table
  timestamps: false // If timestamps are not needed
});

module.exports = Driver;
